src/ORM/DataList.php

```diff
@@ -2187,13 +2190,13 @@ public function chunkedFetch(int $chunkSize = 1000): iterable
     /**
      * Prepopulate any extension caches with the current dataclass and IDs of records
      *
-     * Note that because this calls column() and may result in other database queries based on
-     * the IDs that returns, this should be called after all filtering, sorting, etc has already
+     * Note that because this calls getIDList() and may result in other database queries based on
+     * the IDs that returns, this should be called after all filtering, etc has already
      * been set for this list.
      */
     public function prepopulateCaches(): void
     {
-        $ids = $this->column('ID');
+        $ids = $this->getIDList();
         $this->extend('onPrepopulateCaches', $ids);
```

Comment:
Just used to prepare an associative array that will be checked against later using explicit IDs as keys in the array, so the sort order isn't important.